corehq/motech/repeaters/management/commands/create_missing_repeat_records.py

```diff
@@ -0,0 +1,213 @@
+            total_count += missing_count + successful_count
+
+        if total_missing_count > 0:
+            stats_per_domain[domain] = {
+                'cases': {
+                    'missing_count': total_missing_count,
+                    'percentage_missing': f'{round((total_missing_count / total_count) * 100, 2)}%',
+                }
+            }
+
+    return stats_per_domain
+
+
+def obtain_missing_case_repeat_records_in_domain(domain, repeaters, case, startdate, enddate):
+    successful_count = 0
+    missing_count = 0
+
+    repeat_records = get_repeat_records_by_payload_id(domain, case.get_id)
+    records_in_daterange = [record for record in repeat_records
+                            if startdate <= record.registered_on.date() <= enddate]
+
+    triggered_repeater_ids_and_counts = {}
+    for record in records_in_daterange:
+        current_count = triggered_repeater_ids_and_counts.get(record.repeater_id, 0)
+        if current_count > 0:
+            triggered_repeater_ids_and_counts[record.repeater_id] += 1
+        else:
+            triggered_repeater_ids_and_counts[record.repeater_id] = 1
+
+    for repeater in repeaters:
+        if isinstance(repeater, Dhis2EntityRepeater):
+            # not dealing with this right now
+            continue
+
+        if repeater.started_at.date() >= enddate:
+            # don't count a repeater that was created after the window we care about
+            continue
+        expected_record_count = number_of_repeat_records_triggered_by_case(case, repeater, startdate, enddate)
+        actual_record_count = triggered_repeater_ids_and_counts.get(repeater.get_id, 0)
+
+        # worry about specifying create vs update vs normal later
+        temp_missing_count = expected_record_count - actual_record_count
+        missing_count += temp_missing_count if temp_missing_count > 0 else 0
+        successful_count += actual_record_count
+
+    return missing_count, successful_count
+
+
+def number_of_repeat_records_triggered_by_case(case, repeater, startdate, enddate):
+    filtered_transactions = []
+    if isinstance(repeater, CreateCaseRepeater):
+        # to avoid modifying CreateCaseRepeater's allowed_to_forward method
+        if repeater._allowed_case_type(case) and repeater._allowed_user(case):
+            filtered_transactions = case.transactions[0:1]
+    elif isinstance(repeater, UpdateCaseRepeater):
+        if repeater.allowed_to_forward(case):
+            filtered_transactions = case.transactions[1:]
+    else:
+        if repeater.allowed_to_forward(case):
+            filtered_transactions = case.transactions
+
+    transactions_in_daterange = [transaction for transaction in filtered_transactions
+                                 if startdate <= get_transaction_date(transaction) <= enddate]
+
+    return len(transactions_in_daterange)
+
+
+def get_transaction_date(transaction):
+    return string_to_utc_datetime(transaction.server_date).date()
```

Comment:
how is this date a string? It's a `DateTimeField` on the Django model and I don't think this code works with couch cases (they don't have a `transactions` property)

Reply:
Hmm I see your point. Would python implicitly convert the `DateTimeField` to a string?

Reply:
I also plan on handling couch domains separately since there are so few left. Running on all potentially impacted domains now and so far only 1 domain has an issue which I would guess is a couch domain.